File: src/jvmMain/java/com/dropbear/ffi/components/CustomPropertiesNative.java
package com.dropbear.ffi.components;

import com.dropbear.NativeEngineLoader;

public class CustomPropertiesNative {
    static {
        NativeEngineLoader.ensureLoaded();
    }

    // properties
    public static native String getStringProperty(long worldHandle, long entityHandle, String label);
    public static native int getIntProperty(long worldHandle, long entityHandle, String label);
    public static native long getLongProperty(long worldHandle, long entityHandle, String label);
    public static native double getFloatProperty(long worldHandle, long entityHandle, String label);
    public static native boolean getBoolProperty(long worldHandle, long entityHandle, String label);
    public static native float[] getVec3Property(long worldHandle, long entityHandle, String label);

    public static native void setStringProperty(long worldHandle, long entityHandle, String label, String value);
    public static native void setIntProperty(long worldHandle, long entityHandle, String label, int value);
    public static native void setLongProperty(long worldHandle, long entityHandle, String label, long value);
    public static native void setFloatProperty(long worldHandle, long entityHandle, String label, double value);
    public static native void setBoolProperty(long worldHandle, long entityHandle, String label, boolean value);
    public static native void setVec3Property(long worldHandle, long entityHandle, String label, float[] value);

}

File: src/nativeMain/kotlin/com/dropbear/CustomProperties.native.kt
package com.dropbear

actual fun CustomProperties.getStringProperty(
    entityHandle: Long,
    label: String
): String? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.getIntProperty(entityHandle: Long, label: String): Int? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.getLongProperty(
    entityHandle: Long,
    label: String
): Long? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.getDoubleProperty(
    entityHandle: Long,
    label: String
): Double? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.getFloatProperty(
    entityHandle: Long,
    label: String
): Float? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.getBoolProperty(
    entityHandle: Long,
    label: String
): Boolean? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.getVec3Property(
    entityHandle: Long,
    label: String
): FloatArray? {
    TODO("Not yet implemented")
}

actual fun CustomProperties.setStringProperty(
    entityHandle: Long,
    label: String,
    value: String
) {
}

actual fun CustomProperties.setIntProperty(
    entityHandle: Long,
    label: String,
    value: Int
) {
}

actual fun CustomProperties.setLongProperty(
    entityHandle: Long,
    label: String,
    value: Long
) {
}

actual fun CustomProperties.setFloatProperty(
    entityHandle: Long,
    label: String,
    value: Double
) {
}

actual fun CustomProperties.setBoolProperty(
    entityHandle: Long,
    label: String,
    value: Boolean
) {
}

actual fun CustomProperties.setVec3Property(
    entityHandle: Long,
    label: String,
    value: FloatArray
) {
}

actual fun customPropertiesExistsForEntity(entityId: EntityId): Boolean {
    TODO("Not yet implemented")
}